src/schema/posts.ts

```diff
@@ -1353,20 +1359,26 @@ export const resolvers: IResolvers<any, Context> = {
     ): Promise<GQLEmptyResponse> => updatePromoteToPublicFlag(ctx, id, null),
     updatePinPost: async (
       _,
-      { id, pinned }: PinPostArgs,
+      { id, pinned, swapWithPostId }: PinPostArgs,
       ctx: Context,
     ): Promise<GQLEmptyResponse> => {
       await ctx.con.transaction(async (manager) => {
         const repo = manager.getRepository(Post);
         const post = await repo.findOneBy({ id });
+        const args = { pinnedAt: pinned ? new Date() : null };
 
         await ensureSourcePermissions(
           ctx,
           post.sourceId,
           SourcePermissions.PostPin,
         );
 
-        await repo.update({ id }, { pinnedAt: pinned ? new Date() : null });
+        if (pinned && swapWithPostId) {
+          const swapWithPost = await repo.findOneBy({ id: swapWithPostId });
+          args.pinnedAt = new Date(swapWithPost.pinnedAt.getTime() + 10 * 1000);
+        }
+
+        await repo.update({ id }, args);
```

Comment:
Just generally would it make more sense to create a separate mutation for swapping? It would make the existing mutation more simple.

Reply:
I don't have a strong preference, felt like this was minor enough to tag along inside the existing mutation, but we can extract it if you prefer

Reply:
I think it would, but I imagine that it would also take some more changes on frontend to use different mutation for that. Up to you, don't think of it as blocking. 👍